src/pe/authenticode.rs

```diff
@@ -0,0 +1,117 @@
+// Reference:
+//   https://learn.microsoft.com/en-us/windows-hardware/drivers/install/authenticode
+//   https://download.microsoft.com/download/9/c/5/9c5b2167-8017-4bae-9fde-d599bac8184a/Authenticode_PE.docx
+
+// Authenticode works by omiting sections of the PE binary from the digest
+// those sections are:
+//   - checksum
+//   - data directory entry for certtable
+//   - certtable
+
+use alloc::{boxed::Box, vec::Vec};
+use core::ops::Range;
+
+use super::PE;
+
+impl PE<'_> {
+    /// [`authenticode_ranges`] returns the various ranges of the binary that are relevant for
+    /// signature.
+    pub fn authenticode_ranges(&self) -> ExcludedAuthenticodeSectionsIter<'_> {
```

Comment:
better to move this to `mod.rs` i think.

Reply:
If I do so, then I need to provide a constructor for `ExcludedAuthenticodeSectionsIter`. I kind of like this ability to split `impl` in multiple submodules, but it's up to you.

Reply:
ok yes i agree